Tests.Core/zTests.Core.csproj

```diff
@@ -0,0 +1,27 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```

Comment:
May be problem is in using Microsoft.NET.Sdk? Try to use "Microsoft.NET.Sdk.WindowsDesktop", as in CalcBinding and WpfExample

Reply:
ooh didn't notice the sdk was different, that solved the problem and it can read the namespaces now.
but all the tests that use UI components are failing for a weird reason,
```
Failed   DoubleQuotesStringErrorTest
Error Message:
 Test method Tests.StringTests.DoubleQuotesStringErrorTest threw exception:
System.InvalidOperationException: The calling thread must be STA, because many UI components require this.
Stack Trace:
    at System.Windows.Input.InputManager..ctor()
   at System.Windows.Input.InputManager.GetCurrentInputManagerImpl()
   at System.Windows.Input.InputManager.get_Current()
   at System.Windows.Input.KeyboardNavigation..ctor()
   at System.Windows.FrameworkElement.FrameworkServices..ctor()
   at System.Windows.FrameworkElement.EnsureFrameworkServices()
   at System.Windows.FrameworkElement..ctor()
   at System.Windows.Controls.Control..ctor()
   at System.Windows.Controls.Primitives.TextBoxBase..ctor()
   at System.Windows.Controls.TextBox..ctor()
   at Tests.BaseSystemTests.StringBindingAssert(Binding calcBinding, INotifyPropertyChanged source, Action sourcePropertySetter1, String targetValue1, Action sourcePropertySetter2, String targetValue2, Dictionary`2 resolvedTypes) in E:\Projects\CalcBinding\Tests\BaseSystemTests.cs:line 54
   at Tests.BaseSystemTests.StringAndObjectBindingAssert(Binding calcBinding, INotifyPropertyChanged source, Action sourcePropertySetter1, String targetValue1, Object objTargetValue1, Action sourcePropertySetter2, String targetValue2, Object objTargetValue2, Dictionary`2 resolvedTypes) in E:\Projects\CalcBinding\Tests\BaseSystemTests.cs:line 34
   at Tests.StringTests.DoubleQuotesStringErrorTest() in E:\Projects\CalcBinding\Tests\StringTests.cs:line 116
```
(the same tests don't fail on .net framework)
as i remember this was fixable by adding `[STAThread] ` above the method, but that didn't help, maybe it's a bug in MSTest ?

Reply:
ooh looks like it's a known bug https://github.com/Microsoft/testfx/issues/551

Reply:
Sorry for delay, ooh, problems don't leave us.. :)
Ok, we have following:
1.) MSTest V2 executes test methods in STA thread for .net full Framework and in MTA thread for .net core framework. We can't override that behaviour in runsettings, because it's by design:

> In Test platform V2 ExecutionThreadApartmentState property default value is MTA for .NET Core and STA for .NET Full. STA value is only supported for .NET Framework. Warning should be shown on trying to set value STA for .NET Core and UAP10.0 frameworks tests.

https://github.com/Microsoft/vstest-docs/blob/master/docs/configure.md#execution-thread-apartment-state

2.) MsTest V2 supports writing custom attributes, derived from TestClass or TestMethod, where we can create STA thread and execute test in necessary thread.
3.) There is a library that provides attributes STATestClass, STATestMethod and it do exactly what we want:

https://github.com/saikrishnav/testfxSTAext/tree/master/src/STAExtensions
4.) Library has problems with executing TestInitialize method in STA thread, but we don't need to create controls in this method, so we can use that library
5.) As zTests and zTests.Core projects share tests code base, zTests project can't be compiled, because MSTest V1 doesn't provide extending of TestClass and TestMethod attributes (its sealed).
6.) So I propose:
configure zTests project to target only on .net 4.0
reference to testfxSTAext packet in zTests.Core and use STATestClass from it. For compiling zTests project use that attribute under #if macro, like this:

#if NET40
    [TestClass]
#else
    [Microsoft.VisualStudio.TestTools.UnitTesting.STAExtensions.STATestClass]
#endif

STATestMethod could be used also, but it required to write more #if macros on each method. STATestClass acts on all test methods in one test class.

Reply:
Worked like a charm! (i modified the .net core project instead of the original one though).
I have found another simple issue though, check below comment